client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/RestService.java

```diff
@@ -526,9 +528,14 @@ private void setBasicAuthRequestHeader(HttpURLConnection connection) {
     String userInfo;
     if (basicAuthCredentialProvider != null
         && (userInfo = basicAuthCredentialProvider.getUserInfo(connection.getURL())) != null) {
-      byte[] userInfoBytes = userInfo.getBytes(StandardCharsets.UTF_8);
-      String authHeader = DatatypeConverter.printBase64Binary(userInfoBytes);
-      connection.setRequestProperty("Authorization", "Basic " + authHeader);
+      try {
+        byte[] userInfoBytes =
+          URLDecoder.decode(userInfo, "UTF-8").getBytes(StandardCharsets.UTF_8);
```

Comment:
We should add some test cases to make our intention clear here.

https://github.com/i7c/schema-registry/blob/master/client/src/test/java/io/confluent/kafka/schemaregistry/client/rest/RestServiceTest.java#L85

Might be easier to make this method @VisibleForTesting than try to implicitly test it like the above does though.